cli/pkg/validate/entity/event.go

```diff
@@ -0,0 +1,176 @@
+package entity
+
+import (
+	"fmt"
+
+	"github.com/rudderlabs/rudder-iac/cli/pkg/localcatalog"
+	"github.com/samber/lo"
+)
+
+var (
+	TrackEventTypes    = []string{"track"}
+	NonTrackEventTypes = []string{"identify", "group", "page", "screen"}
+)
+
+var (
+	_ TypedCatalogEntityValidator[*localcatalog.Event] = &EventValidator{}
+	_ ValidationRule[*localcatalog.Event]              = &EventRequiredKeysRule{}
+	_ ValidationRule[*localcatalog.Event]              = &EventDuplicateKeysRule{}
+)
+
+type EventValidator struct {
+	rules []ValidationRule[*localcatalog.Event]
+}
+
+func (ev *EventValidator) RegisterRule(rule ValidationRule[*localcatalog.Event]) {
+	ev.rules = append(ev.rules, rule)
+}
+
+func (ev *EventValidator) Validate(dc *localcatalog.DataCatalog) (errs []ValidationError) {
+	log.Info("validating event entities in the catalog")
+
+	for group, events := range dc.Events {
+		for _, event := range events {
+
+			reference := fmt.Sprintf(
+				"#/events/%s/%s",
+				group,
+				event.LocalID,
+			)
+
+			for _, rule := range ev.rules {
+				errs = append(errs, rule.Validate(reference, event, dc)...)
+			}
+		}
+	}
+
+	return errs
+}
+
+type EventRequiredKeysRule struct {
```

Comment:
Given you take a mostly functional approach here, and have all logic in a validation functions (which is great), can we avoid these empty structs? Can we have a function signature instead and pass functions directly?